modules/kubernetes-addons/locals.tf

```diff
@@ -19,4 +19,10 @@ locals {
     kubeStateMetrics          = var.enable_kube_state_metrics ? module.kube_state_metrics[0].argocd_gitops_config : null
     kubernetesDashboard       = var.enable_kubernetes_dashboard ? module.kubernetes_dashboard[0].argocd_gitops_config : null
   }
+
```

Comment:
We can get values for each lookup in locals file
e.g.,
```hcl
  account_id        = data.aws_caller_identity.current.account_id
  aws_partition     = data.aws_partition.current.id
  ...
```

This will be used while calling the add-ons. Something like below

```hcl
module "ingress_nginx" {
  account_id        = local.account_id
  aws_partition     = local.aws_partition
}
```

Reply:
Agree, but shouldn't we prefer to send over the full result, and each consumer can decide which attribute to use from it (`account_id` /`id`/`user_id` etc) ?
Following your comment below about the actual values, we can go that direction, and that might make the `context` object bigger as we'll choose each attribute/value for every data source we may need.

There's tradeoff to each decision, but I do tend to go with what you suggest for now, will work on it!